<launch>

    <arg name="map"/>
    <arg name="db_path" default="/home/nbore/mongodb_store"/>
    <arg name="test" default="false"/>
    <arg name="number_targets" default="2"/>

    <!--
    <include file="$(find mongodb_store)/launch/mongodb_store.launch">
        <param name="db_path" value="$(arg db_path)"/>
        <param name="port" value="62345"/>
    </include>
    -->

    <node pkg="map_server" type="map_server" name="map_server" args="$(arg map)"/>

    <node pkg="rbpf_mtt" type="test_map_server.py" name="test_map_server" output="screen" respawn="true" if="$(arg test)">
        <param name="number_targets" value="$(arg number_targets)"/>
    </node>
    <node pkg="rbpf_mtt" type="test_filter.py" name="test_filter" output="screen" respawn="true" unless="$(arg test)">
        <param name="number_targets" value="$(arg number_targets)"/>
    </node>

    <node pkg="rbpf_mtt" type="measurement_simulator" name="measurement_simulator" output="screen" respawn="true">
        <param name="number_targets" value="$(arg number_targets)"/>
    </node>

    <node pkg="rbpf_mtt" type="gmm_map_server" name="gmm_map_server" output="screen" respawn="true">
        <param name="number_targets" value="$(arg number_targets)"/>
    </node>

    <!-- <node pkg="rbpf_mtt" type="test_filter.py" name="test_filter"/> -->

    <!-- Service: configs -->
    <!-- <node= pkg="soma_roi" type="soma_roi_drawer.py" name="soma_roi_drawer"/> -->

    <!-- Query ROI service -->
    <!-- <node= pkg="soma_roi" type="soma_manager.py" name="soma_manager"/> -->

    <!-- <node= pkg="soma_roi" type="soma_map_manager.py" name="soma_map_manager"/> -->

</launch>
